cli/commands.go

```diff
@@ -43,69 +42,63 @@ func CheckTools(c *cli.Context) error {
 }
 
 // CreateCluster creates a new single-node cluster container and initializes the cluster directory
-func CreateCluster(c *cli.Context) error {
-
-	if err := CheckClusterName(c.String("name")); err != nil {
+func CreateCluster(
+	name,
+	image string,
+	oEnv []string,
+	port string,
+	serverArgs, agentArgs []string,
+	workers int,
+	publish, volumes []string,
+	autoRestart bool,
+	portAutoOffset, wait int,
+) error {
+
+	if err := CheckClusterName(name); err != nil {
 		return err
 	}
 
-	if cluster, err := getClusters(false, c.String("name")); err != nil {
+	if cluster, err := getClusters(false, name); err != nil {
 		return err
 	} else if len(cluster) != 0 {
 		// A cluster exists with the same name. Return with an error.
-		return fmt.Errorf(" Cluster %s already exists", c.String("name"))
+		return fmt.Errorf(" Cluster %s already exists", name)
 	}
 
 	// On Error delete the cluster.  If there createCluster() encounter any error,
 	// call this function to remove all resources allocated for the cluster so far
 	// so that they don't linger around.
 	deleteCluster := func() {
-		if err := DeleteCluster(c); err != nil {
-			log.Printf("Error: Failed to delete cluster %s", c.String("name"))
+		if err := DeleteCluster(false, name); err != nil {
+			log.Printf("Error: Failed to delete cluster %s", name)
 		}
 	}
 
 	// define image
-	image := c.String("image")
-	if c.IsSet("version") {
-		// TODO: --version to be deprecated
-		log.Warning("The `--version` flag will be deprecated soon, please use `--image rancher/k3s:<version>` instead")
-		if c.IsSet("image") {
-			// version specified, custom image = error (to push deprecation of version flag)
-			log.Fatalln("Please use `--image <image>:<version>` instead of --image and --version")
-		} else {
-			// version specified, default image = ok (until deprecation of version flag)
-			image = fmt.Sprintf("%s:%s", strings.Split(image, ":")[0], c.String("version"))
-		}
-	}
```

Comment:
I want to bring attention to the fact that I removed this bit that is marked to be deprecated.